src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/CpxVariantCanonicalRepresentation.java

```diff
@@ -168,8 +171,14 @@ static SimpleInterval getAffectedReferenceRegion(final List<SimpleInterval> even
             // there shouldn't be a segment constructed if two segmenting locations are adjacent to each other on the reference
             // this could happen when (in the simplest case), two alignments are separated by a mapped insertion (hence 3 total alignments),
             // and the two alignments' ref span are connected
+            // more generally: only segment when the two segmenting locations are boundaries of alignments that overlap each other (ref. span)
             if (rightBoundary.getStart() - leftBoundary.getEnd() > 1) {
                 segments.add(new SimpleInterval(eventPrimaryChromosome, leftBoundary.getStart(), rightBoundary.getStart()));
+            } else if (rightBoundary.getStart() - leftBoundary.getEnd() == 1) {
+                final SimpleInterval twoBaseSegment = new SimpleInterval(eventPrimaryChromosome, leftBoundary.getEnd(), rightBoundary.getStart());
+                if ( twoBaseBoundaries.contains(twoBaseSegment) ) {
+                    segments.add(twoBaseSegment);
+                }
```

Comment:
Assuming I'm correct about the above being a typo, you might decrease scope for this kind of error be refactoring thusly:
final int start = rightBoundary.getStart();
final int end = leftBoundary.getEnd();
final int segmentLength = end - start;
if( segmentLength >= 1) {
    final SimpleInterval newSegment = new SimpleInterval(eventPrimaryChromosome, start, end);
    if(segmentLength > 1 || twoBaseBoundaries.contains(newSegment)) {
        segments.add(newSegment);
    }
}

Reply:
I see what you mean, but like I replied in the other place, it's a "point" location so won't matter in this case.
And the suggestion you made is good, except that end is always >= start because it is from the "left" (I know, so many mind-numbing details...)